src/plugins/d3/renderer/hooks/useShapes/treemap/index.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import {
+    color,
+    pointer,
+    select,
+    treemap,
+    treemapBinary,
+    treemapDice,
+    treemapSlice,
+    treemapSliceDice,
+    treemapSquarify,
+} from 'd3';
+import type {BaseType, Dispatch, HierarchyRectangularNode} from 'd3';
+import get from 'lodash/get';
+
+import {LayoutAlgorithm} from '../../../../../../constants';
+import type {TooltipDataChunkTreemap} from '../../../../../../types';
+import {setEllipsisForOverflowTexts} from '../../../utils';
+import {block} from '../../../../../../utils/cn';
+
+import {PreparedSeriesOptions} from '../../useSeries/types';
+import type {PreparedTreemapData, PreparedTreemapSeriesData, TreemapLabelData} from './types';
+import {getLabelData} from './utils';
+
+const b = block('d3-treemap');
+const DEFAULT_PADDING = 1;
+
+type ShapeProps = {
+    dispatcher: Dispatch<object>;
+    preparedData: PreparedTreemapData;
+    seriesOptions: PreparedSeriesOptions;
+    svgContainer: SVGSVGElement | null;
+    width: number;
+    height: number;
+};
+
+export const TreemapSeriesShape = (props: ShapeProps) => {
+    const {dispatcher, preparedData, seriesOptions, svgContainer, width, height} = props;
+    const ref = React.useRef<SVGGElement>(null);
+
+    React.useEffect(() => {
+        if (!ref.current) {
+            return () => {};
+        }
+
+        const svgElement = select(ref.current);
+        svgElement.selectAll('*').remove();
+        const {hierarchy, series} = preparedData;
+        const treemapInstance = treemap<PreparedTreemapSeriesData>();
+
+        switch (series.layoutAlgorithm) {
+            case LayoutAlgorithm.Binary: {
+                treemapInstance.tile(treemapBinary);
+                break;
+            }
+            case LayoutAlgorithm.Dice: {
+                treemapInstance.tile(treemapDice);
+                break;
+            }
+            case LayoutAlgorithm.Slice: {
+                treemapInstance.tile(treemapSlice);
+                break;
+            }
+            case LayoutAlgorithm.SliceDice: {
+                treemapInstance.tile(treemapSliceDice);
+                break;
+            }
+            case LayoutAlgorithm.Squarify: {
+                treemapInstance.tile(treemapSquarify);
+                break;
+            }
+        }
+
+        const root = treemapInstance.size([width, height]).paddingInner((d) => {
+            const levelOptions = series.levels?.find((l) => l.index === d.depth + 1);
+            return levelOptions?.padding ?? DEFAULT_PADDING;
+        })(hierarchy.sum((d) => d.value || 0));
```

Comment:
Perhaps this should be moved to `prepare-data`? So that only the rendering of the prepared data remains here